### ISSUE TITLE
CommitDialog: diff of wrong file is showed

### ISSUE THREAD

**Issue by DmitryZhelnin** [gitextensions/gitextensions#6561]
## Current behaviour

Sometimes when I select staged file in the CommitDialog and then select unstaged file, diff is shown for the first unstaged file instead of the selected one.

## Expected behaviour

Diff for the selected file is shown.

## Steps to reproduce
1. Make changed to several files.
2. Open CommitDialog and stage two of more files.
3. Leave some files unstaged.
4. Select staged file, look at the diff.
5. Select non first unstaged file, look at the diff.
6. Sometimes the diff of the first unstaged file is shown. If not - go to 4.

## Did this work in previous version of GitExtensions

Didn't face this issue in the latest release version 3.0.2.

## Environment

- Git Extensions 3.1.0.5766-RC1
- Build 497059b2205086d8a44a33dd29f2cae2b4bfffc2 (Dirty)
- Git 2.21.0.windows.1
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.3324.0
- DPI 120dpi (125% scaling)

**Comment by DmitryZhelnin** [comment]
If you click an empty area in the unstaged files list the first file gets selected and deselected

![commit_dialog](https://user-images.githubusercontent.com/4993430/57694697-11890180-7655-11e9-8ac0-64e1a3e15ad5.gif)

**Comment by LogicDaemon** [comment]
+1

![2019-05-15 GitExtensions fail to display proper file in Commit window](https://user-images.githubusercontent.com/7845494/57771899-c803e980-771c-11e9-83eb-e19a917e643d.gif)

**Comment by RussKie** [comment]
Is this the same as #5213?

**Comment by DmitryZhelnin** [comment]
> 
> 
> Is this the same as #5213?

Yes, I believe I've created a duplicate.

**Comment by LogicDaemon** [comment]
> Is this the same as #5213?

I don't think so, the #5213 says «A diff for the first file is shown **for a moment** and after that a diff for the selected file is shown.» In my case, it never switches to selected file on it's own if it's not a change but an added file, I have to select some other file and back.

Also, double-clicking any file adds first file to the stage instead of clicked one.
